README.md

```diff
@@ -49,4 +49,31 @@ using (var stream = current.File("test.txt").Create())
 //create a "test.txt" file without extension
 using (var stream = current.FileSystem.FileInfo.FromFileName(current.FileSystem.Path.Combine(current.FullName, "test.txt")).Create())
     stream.Dispose();
+```
+
+## Automatic cleanup with Disposable extensions
+
+```csharp
+var fs = new FileSystem();
+
+//with extension
+using (fs.CreateDisposableDirectory(out IDirectoryInfo dir))
+{
+    Console.WriteLine($"This directory will be deleted when control leaves the using block: '{dir.FullName}'");
+}
+
+//without extension
+var temp = fileSystem.Path.GetTempPath();
+var fileName = fileSystem.Path.GetRandomFileName();
+var path = fileSystem.Path.Combine(temp, fileName);
+
+try
+{
+    IDirectoryInfo dir = fs.Directory.CreateDirectory(path);
+    Console.WriteLine($"This directory will be deleted in the finally block: '{dir.FullName}'");
+}
+finally
+{
+    fs.Directory.Delete(path, recursive: true);
+}
 ```
```

Comment:
can you please also add an example for CreateDisposableFile.
or maybe just say the same pattern above applies for files with the call CreateDisposableFile

Reply:
I went the route of adding a bit of text before the code block to note that the same pattern applies to files as well. Let me know if looks good to you!